src/DynamoCoreWpf/ViewModels/Menu/PreferencesViewModel.cs

```diff
@@ -1586,6 +1586,16 @@ public void ResetBackupLocation()
             dynamoViewModel.Model.ResetPreferenceItemLocation(Core.PathManager.PreferenceItem.Backup);
         }
 
+        /// <summary>
+        ///     Event that is fired when a node is removed from the workspace.
+        /// </summary>
+        public event Action PreferencesChanged;
```

Comment:
I'm not sure about this one. 
We are calling this from the onClose but there is no guarantee that the preferences have actually changed.
Plus the underlying object we care about is the PreferenceSettings instance and one can subscribe to that.
However the Preferences dlg is modal so we should only do the update onClose.

But I don't think we need to do the real time update so I'm ok with removing this. We only consider the options on first occasion like the flyout show.

Reply:
Its weirder if its not realtime as the visual state gets desynchronized with the internal state. Its fine if we reload the view each time the user opens and closes the preferences, which shouldn't happen too often while the flyout is still visible.

We can have the best of both if we introduce an object to represent the latest copy of the preferences settings we're interested in and comparing with it, but it could introduce bugs when modifying the code and not updating that object.

So, I'd prefer it to load every time instead of get bugged when not refreshing. Plus, this may be useful for other parts of the application that would benefit from dynamically updating.

Reply:
Let's rename it then OnPreferencesClosed or something and maybe make it internal.

Reply:
Can't make it internal as we're using it from another assembly, but we can rename it

Reply:
We can access internals from our assembly since we have this [assembly: InternalsVisibleTo("NodeAutoCompleteViewExtension")].

But thinking more about it maybe we can avoid it all together and be more inline with the way other settings are handled.
So how about we define out DNA event in preferences for all the DNA settings and subscribe/react to it from the extension as usual.
The preferences is modal so we'll just need to postpone the action on Iddling - we do this in other contexts as well.

Reply:
Updated